src/DataSource/Types.hs

```diff
@@ -454,6 +436,29 @@ instance FromJSON Attachment where
             , aSize         = attachmentSize
             }
 
+-- | JSON Parsing
+instance FromJSON Comment where
+    parseJSON = withObject "comment" $ \o -> do
+        commentId           <- o .: "id"
+        commentBody         <- o .: "body"
+        commentAttachments  <- o .: "attachments"
+        commentIsPublic     <- o .: "public"
+        commentAuthorId     <- o .: "author_id"
+
+        pure Comment
+            { cId          = commentId
+            , cBody        = commentBody
+            , cAttachments = commentAttachments
+            , cPublic      = commentIsPublic
+            , cAuthor      = commentAuthorId
+            }
+
+-- Is there an way to make this better?
+instance (FromJSON a) => FromJSON (PageResultList a) where
+    parseJSON = withObject "ticketList" $ \o ->
+            PageResultList
+                <$> (o .: "tickets" <|>  o .: "users")
+                <*> o .: "next_page"
```

Comment:
This does work but I feel like this is naive way to implement it.. Any better idea?